docs/references/config/config.toml.md

```diff
@@ -0,0 +1,2055 @@
+```
+
+| Value type          | string                                          |
+|:--------------------|:------------------------------------------------|
+| **Possible values** | relative directory path, appended to `$CMTHOME` |
+|                     | absolute directory path                         |
+
+The default relative path translates to `$CMTHOME/data/cs.wal/wal`. In case `$CMTHOME` is unset, it defaults to
+`$HOME/.cometbft/data/cs.wal/wal`.
+
+The consensus WAL stores all consensus messages received and broadcast by a
+node, as well as some important consensus events (e.g., new height and new round step).
+The goal of this log is to enable a node that crashes and late recovers
+to re-join consensus with the same state it has before crashing.
+Recovering nodes that "forget" the actions taken before crashing are faulty
+nodes that are likely to present Byzantine behavior (e.g., double signing).
+
+### consensus.timeout_propose
+
+How long a node waits for the proposal block before prevoting nil.
+
+```toml
+timeout_propose = "3s"
+```
+
+| Value type          | string (duration) |
+|:--------------------|:------------------|
+| **Possible values** | &gt;= `"0s"`      |
+
+The proposal block of a round of consensus is broadcast by the proposer of that round.
+The `timeout_propose` should be large enough to encompass the common-case
+propagation delay of a `Proposal` and one or more `BlockPart` (depending on the
+proposed block size) messages from any validator to the node.
+
+If the proposed block is not received within `timeout_propose`, validators
+issue a prevote for nil, indicating that they have not received, and
+therefore are unable to vote for, the block proposed in that round.
+
+Setting `timeout_propose` to `0s` means that the validator does not wait at all
+for the proposal block and always prevotes nil.
+This has obvious liveness implications since this validator will never prevote
+for proposed blocks.
+
+### consensus.timeout_propose_delta
+
+How much `timeout_propose` increases with each round.
+
+```toml
+timeout_propose_delta = "500ms"
+```
+
+| Value type          | string (duration) |
+|:--------------------|:------------------|
+| **Possible values** | &gt;= `"0ms"`     |
+
+Consensus timeouts are adaptive.
+This means that when a round of consensus fails to commit a block, the next
+round of consensus will adopt increased timeout durations.
+Timeouts increase linearly over rounds, so that the `timeout_propose` adopted
+in round `r` is `timeout_propose + r * timeout_propose_delta`.
+
+### consensus.timeout_prevote
+
+How long a node waits, after receiving +2/3 conflicting prevotes, before pre-committing nil.
+
+```toml
+timeout_prevote = "1s"
+```
+
+| Value type          | string (duration) |
+|:--------------------|:------------------|
+| **Possible values** | &gt;= `"0s"`      |
+
+A validator that receives +2/3 prevotes for a block, precommits that block.
+If it receives +2/3 prevotes for nil, it precommits nil.
```

Comment:
@cason shouldn't this read +1/3 ?

Reply:
Why? It is true that it precommits nil if no value reaches 2/3+ prevotes. But this is described in the next sentence.

Reply:
I mean, this could be an optimization, but the code isn't like that.

Reply:
I see. That was my question, which is now answered (--> it's a possible optimization). Resolving.